From 0948a9c5386095baae4012190a6b65aba684a907 Mon Sep 17 00:00:00 2001
Message-Id: <0948a9c5386095baae4012190a6b65aba684a907.1659925856.git.geliang.tang@suse.com>
From: James Smart <jsmart2021@gmail.com>
Date: Fri, 1 Jul 2022 14:14:17 -0700
Subject: [PATCH] scsi: lpfc: Remove extra atomic_inc on cmd_pending in
 queuecommand after VMID
Git-commit: 0948a9c5386095baae4012190a6b65aba684a907
Patch-mainline: v6.0-rc1
References: bsn#210
Modified-by-SEL: Yes, refreshed due to context changes

VMID introduced an extra increment of cmd_pending, causing double-counting
of the I/O. The normal increment ios performed in lpfc_get_scsi_buf.

Link: https://lore.kernel.org/r/20220701211425.2708-5-jsmart2021@gmail.com
Fixes: 33c79741deaf ("scsi: lpfc: vmid: Introduce VMID in I/O path")
Cc: <stable@vger.kernel.org> # v5.14+
Co-developed-by: Justin Tee <justin.tee@broadcom.com>
Signed-off-by: Justin Tee <justin.tee@broadcom.com>
Signed-off-by: James Smart <jsmart2021@gmail.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/scsi/lpfc/lpfc_scsi.c | 1 -
 1 file changed, 1 deletion(-)

--- a/drivers/scsi/lpfc/lpfc_scsi.c
+++ b/drivers/scsi/lpfc/lpfc_scsi.c
@@ -5754,7 +5754,6 @@ lpfc_queuecommand(struct Scsi_Host *shos
 		}
 	}
 
-	atomic_inc(&ndlp->cmd_pending);
 #ifdef CONFIG_SCSI_LPFC_DEBUG_FS
 	if (unlikely(phba->hdwqstat_on & LPFC_CHECK_SCSI_IO))
 		this_cpu_inc(phba->sli4_hba.c_stat->xmt_io);
